src/services/concretes/state.service.ts

```diff
@@ -0,0 +1,72 @@
+import { DataAccessLayer } from "@blockr/blockr-data-access";
+import { State, Transaction } from "@blockr/blockr-models";
+import { inject, injectable } from "inversify";
+import { ConstantStore } from "../../stores";
+
+@injectable()
+export class StateService {
+    private readonly constantStore: ConstantStore;
+    private readonly dataAccessLayer: DataAccessLayer;
+
+    constructor(@inject(ConstantStore) constantStore: ConstantStore,
+                @inject(DataAccessLayer) dataAccessLayer: DataAccessLayer) {
+        this.constantStore = constantStore;
+        this.dataAccessLayer = dataAccessLayer;
+    }
+    
+    public async updateStatesForTransactionsAsync(transactions: Transaction[]): Promise<State[]> {
+        return new Promise(async (resolve, reject) => {
+            try {
+                const states: State[] = [];
+
+                for (const transaction of transactions) {
+                    let senderState = await this.dataAccessLayer
+                        .getStateAsync(transaction.transactionHeader.senderKey);
+                    let recipientState = await this.dataAccessLayer
+                        .getStateAsync(transaction.transactionHeader.recipientKey);
+
+                    senderState = this.updateStateAmount(
+                        "-",
```

Comment:
I'd prefer you to change the string to an enum. You could consider keeping the enum privately within this class if that's possible.